src/core/ext/filters/client_channel/resolver/dns/event_engine/event_engine_client_channel_resolver.cc

```diff
@@ -0,0 +1,513 @@
+        addresses_.emplace_back(CreateGRPCResolvedAddress(addr),
+                                resolver_->channel_args());
+      }
+    }
+    result = OnResolvedLocked();
+  }
+  if (result.has_value()) {
+    resolver_->OnRequestComplete(std::move(*result));
+  }
+}
+
+void EventEngineClientChannelDNSResolver::EventEngineDNSRequestWrapper::
+    OnSRVResolved(
+        absl::StatusOr<std::vector<EventEngine::DNSResolver::SRVRecord>>
+            srv_records) {
+  ValidationErrors::ScopedField field(&errors_, "srv lookup");
+  absl::optional<Resolver::Result> result;
+  auto cleanup = absl::MakeCleanup([&]() {
+    if (result.has_value()) {
+      resolver_->OnRequestComplete(std::move(*result));
+    }
+  });
+  MutexLock lock(&on_resolved_mu_);
+  if (orphaned_) return;
+  srv_handle_.reset();
+  if (!srv_records.ok()) {
+    // An error has occurred, finish resolving.
+    errors_.AddError(srv_records.status().message());
+    result = OnResolvedLocked();
+    return;
+  }
+  // Do a subsequent hostname query since SRV records were returned
+  for (auto& srv_record : *srv_records) {
+    GRPC_EVENT_ENGINE_RESOLVER_TRACE(
+        "DNSResolver::%p Started resolving balancer hostname %s:%d",
+        resolver_.get(), srv_record.host.c_str(), srv_record.port);
+    balancer_hostname_handles_.insert(event_engine_resolver_->LookupHostname(
+        [host = std::move(srv_record.host),
+         self = Ref(DEBUG_LOCATION, "OnBalancerHostnamesResolved")](
+            absl::StatusOr<std::vector<EventEngine::ResolvedAddress>>
+                new_balancer_addresses) mutable {
+          self->OnBalancerHostnamesResolved(std::move(host),
+                                            std::move(new_balancer_addresses));
+        },
+        srv_record.host, std::to_string(srv_record.port),
+        resolver_->query_timeout_ms_));
+  }
+}
+
+void EventEngineClientChannelDNSResolver::EventEngineDNSRequestWrapper::
+    OnBalancerHostnamesResolved(
+        std::string authority,
+        absl::StatusOr<std::vector<EventEngine::ResolvedAddress>>
+            new_balancer_addresses) {
+  ValidationErrors::ScopedField field(
+      &errors_,
+      absl::StrCat("balancer hostname lookup, authority: ", authority));
+  absl::optional<Resolver::Result> result;
+  auto cleanup = absl::MakeCleanup([&]() {
+    if (result.has_value()) {
+      resolver_->OnRequestComplete(std::move(*result));
+    }
+  });
+  MutexLock lock(&on_resolved_mu_);
+  if (orphaned_) return;
+  ++number_of_balancer_hostnames_resolved_;
+  if (!new_balancer_addresses.ok()) {
+    // An error has occurred, finish resolving.
+    errors_.AddError(new_balancer_addresses.status().message());
+  } else {
+    // Capture the addresses and finish resolving.
+    balancer_addresses_.reserve(balancer_addresses_.size() +
+                                new_balancer_addresses->size());
+    auto srv_channel_args =
+        resolver_->channel_args().Set(GRPC_ARG_DEFAULT_AUTHORITY, authority);
```

Comment:
This should start with an empty ChannelArgs object rather than using `resolver_->channel_args()`.

Reply:
Done.